lib/jobs/job.go

```diff
@@ -0,0 +1,79 @@
+package jobs
+
+import (
+	"fmt"
+	"strings"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/kosyfrances/rundeck-zabbix/lib"
+	"github.com/kosyfrances/rundeck-zabbix/lib/zabbix"
+)
+
+type nodeFilters struct {
+	Filter string `yaml:"filter"`
+}
+
+type commands struct {
+	Exec string `yaml:"exec"`
+}
+
+type sequence struct {
+	Commands []commands `yaml:"commands"`
+}
+
+type job struct {
+	Name        string      `yaml:"name"`
+	Description string      `yaml:"description"`
+	NodeFilters nodeFilters `yaml:"nodefilters"`
+	Sequence    sequence    `yaml:"sequence"`
+}
+
+/*
+Make generates Rundeck job model document in Yaml format
+with triggers' information from Zabbix
+
+The file output similar to:
+
+- name: test-job
+  description: Just a random test
+  nodefilters:
+    filter: 'dummy-host'
+  sequence:
+    commands:
+    - exec: ""
+*/
+func Make(results zabbix.TriggerResults, file, prefix string) error {
+
+	var jobList []job
+
+	for _, result := range results {
+		if strings.HasPrefix(result.Description, prefix) {
+			for _, host := range result.Hosts {
+				resultFilter := nodeFilters{
+					Filter: host.Name,
+				}
+
+				seq := sequence{
+					Commands: []commands{
```

Comment:
What's the reason for initializing sequence this way?
This can be written as  

    Commands: []commands{commands{}} 

or when renamed 

    Command: []command{command{}}

Reply:
I will go with `Commands: []commands{commands{}} `. This is just a place holder to hold a list of commands which the user will add to the file.